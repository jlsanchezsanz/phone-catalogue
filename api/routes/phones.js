const phonesRoutes = (app, fs) => {
  const dataPath = './data/phones.json';

  app.get('/phones', (_, res) => {
    fs.readFile(dataPath, 'utf8', (err, data) => {
      setTimeout(() => {
        if (err) {
          throw err;
        }
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.send(JSON.parse(data));
      }, 3000);
    });
  });
};

module.exports = phonesRoutes;
