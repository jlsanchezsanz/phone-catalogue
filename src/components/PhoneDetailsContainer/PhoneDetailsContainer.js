import React from 'react';
import { connect } from 'react-redux';

import PhoneDetails from './PhoneDetails';

function PhoneDetailsContainer(props) {
  return !props.phone ? (
    <p>Loading...</p>
  ) : (
    <PhoneDetails phone={props.phone} />
  );
}

const mapStateToProps = (state, ownProps) => ({
  phone: state.phonesReducer.phones.find(
    (phone) => phone.id.toString() === ownProps.match.params.id
  ),
  isLoading: state.phonesReducer.isLoading
});

export default connect(mapStateToProps)(PhoneDetailsContainer);
