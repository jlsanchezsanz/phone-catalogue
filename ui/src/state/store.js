import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import RootReducer from './reducers';

export const initialState = {};

export const middlewares = [ReduxThunk];

export const store = createStore(
  RootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middlewares))
);
