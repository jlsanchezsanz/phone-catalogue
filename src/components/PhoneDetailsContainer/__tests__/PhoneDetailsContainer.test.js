import React from 'react';
import { mount } from 'enzyme';

import PhoneDetailsContainer from '../PhoneDetailsContainer';
import { mockStore } from '../../../state/__mocks__/store';
import { phonesMock } from '../../../mocks/phones.mock';

jest.mock('../PhoneDetails', () => () => <></>);

const setUpMount = (initialState, match) => {
  const store = mockStore(initialState);
  const component = mount(
    <PhoneDetailsContainer store={store} match={match} />
  );
  return component;
};

describe('PhoneDetailsContainer', () => {
  let component;

  it('should display loading message while no phone', () => {
    component = setUpMount(
      { phonesReducer: { phones: phonesMock.data } },
      { params: {} }
    );
    expect(component).toMatchSnapshot();
  });

  it('should display phone details', () => {
    component = setUpMount(
      { phonesReducer: { phones: phonesMock.data } },
      { params: { id: '1' } }
    );
    expect(component).toMatchSnapshot();
  });
});
