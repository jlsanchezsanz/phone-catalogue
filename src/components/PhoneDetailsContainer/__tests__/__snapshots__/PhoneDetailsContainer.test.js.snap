// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetailsContainer should display loading message while no phone 1`] = `
<Connect(PhoneDetailsContainer)
  match={
    Object {
      "params": Object {},
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PhoneDetailsContainer
    dispatch={[Function]}
    match={
      Object {
        "params": Object {},
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <p>
      Loading...
    </p>
  </PhoneDetailsContainer>
</Connect(PhoneDetailsContainer)>
`;

exports[`PhoneDetailsContainer should display phone details 1`] = `
<Connect(PhoneDetailsContainer)
  match={
    Object {
      "params": Object {
        "id": "1",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PhoneDetailsContainer
    dispatch={[Function]}
    match={
      Object {
        "params": Object {
          "id": "1",
        },
      }
    }
    phone={
      Object {
        "color": "grey",
        "description": "Introducing the smartphone your life can not do without, The Samsung Galaxy S7. Packed with features to keep you both productive and entertained, all in a sleek, slim design that fits comfortably in your hand or pocket.",
        "id": 1,
        "images": Array [
          "Galaxy_S7.png",
        ],
        "manufacturer": "Samsung",
        "name": "Galaxy S7",
        "price": 209,
        "processor": "Snapdragon 820",
        "ram": 4,
        "screen": "5,1 inch Quad-HD",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Component
      phone={
        Object {
          "color": "grey",
          "description": "Introducing the smartphone your life can not do without, The Samsung Galaxy S7. Packed with features to keep you both productive and entertained, all in a sleek, slim design that fits comfortably in your hand or pocket.",
          "id": 1,
          "images": Array [
            "Galaxy_S7.png",
          ],
          "manufacturer": "Samsung",
          "name": "Galaxy S7",
          "price": 209,
          "processor": "Snapdragon 820",
          "ram": 4,
          "screen": "5,1 inch Quad-HD",
        }
      }
    />
  </PhoneDetailsContainer>
</Connect(PhoneDetailsContainer)>
`;
